### PR TITLE
Moved some more unit tests to Codeception

### PR DESCRIPTION
**Pull request by SidRoberts**

### CODE REVIEW

File: tests/unit/Cli/RouterTest.php [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+ * so that we can send you a copy immediately.
+ */
+class RouterTest extends UnitTest
+{
+    public function testRouters()
+    {
+        $this->specify(
+            "...",
+            function () {
+                $di = new \Phalcon\DI\FactoryDefault\CLI();
```
Comment (sergeyklay) [inline tests/unit/Cli/RouterTest.php]: 1. Could you please use aliasing like `use Phalcon\Di\FactoryDefault\Cli?
2. Note `Di` not `DI` and `Cli` not `CLI`
3. Provide please real failed message  instead `"..."`

File: tests/unit/Cli/RouterTest.php [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
+                        return "data";
+                    }
+                );
+
+                $router = new \Phalcon\CLI\Router();
+
+                $router->handle([]);
+                expect($router->getModuleName())->equals(null);
+                expect($router->getTaskName())->equals(null);
+                expect($router->getActionName())->equals(null);
```
Comment (sergeyklay) [inline tests/unit/Cli/RouterTest.php]:
```php
expect($router->getModuleName())->null();
expect($router->getTaskName())->null();
expect($router->getActionName())->null();
```

File: tests/unit/Cli/RouterTest.php [hunk 3]
```diff
@@ -0,0 +1,728 @@
...
+                        'params' => array('id' => '150')
+                    ),
+                    array(
+                        'uri' => 'very static route',
+                        'module' => null,
+                        'task' => 'static',
+                        'action' => 'route',
+                        'params' => array()
+                    ),
+                );
```
Comment (sergeyklay) [inline tests/unit/Cli/RouterTest.php]: Use `examples` please: https://github.com/Codeception/Specify#examples

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
@SidRoberts Could you please take a look at my comments :)

File: tests/unit/Cli/ConsoleTest.php [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+
+/**
+ * \Phalcon\Test\Unit\Cli\ConsoleTest
+ * Tests the \Phalcon\Cli\Console component
+ *
+ * @copyright (c) 2011-2016 Phalcon Team
+ * @link      http://www.phalconphp.com
+ * @author    Andres Gutierrez <andres@phalconphp.com>
+ * @author    Nikolaos Dimopoulos <nikos@phalconphp.com>
+ * @package   Phalcon\Test\Unit
```
Comment (sergeyklay) [inline tests/unit/Cli/ConsoleTest.php]: `Phalcon\Test\Unit\Cli`

File: tests/unit/Cli/ConsoleTest.php [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+
+                expect($console->getModules())->equals($expected);
+            }
+        );
+    }
+
+    public function testIssue787()
+    {
+        $this->specify(
+            "...",
```
Comment (sergeyklay) [inline tests/unit/Cli/ConsoleTest.php]: provide message

File: tests/unit/Cli/ConsoleTest.php [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+                            'World',
+                            '######'
+                        ]
+                    );
+                    expect($dispatcher->getTaskName())->equals('main');
+                    expect($dispatcher->getActionName())->equals('hello');
+                    expect($dispatcher->getParams())->equals(array('World', '######'));
+                    expect($dispatcher->getReturnedValue())->equals('Hello World######');
+                } catch (\Exception $e) {
+                    expect($e->getMessage())->equals("Module 'devtools' isn't registered in the console container");
```
Comment (sergeyklay) [inline tests/unit/Cli/ConsoleTest.php]: Split please this test to 2 tests: the testing module and the testing namespace.

Also note if exception will not be thrown, the test will pass. This is why you have to use something like this: https://github.com/Codeception/Specify#exceptions

File: tests/unit/Cli/ConsoleTest.php [hunk 7]
```diff
@@ -0,0 +1,490 @@
...
+                            'World',
+                            '!'
+                        ]
+                    );
+                    expect($dispatcher->getTaskName())->equals('main');
+                    expect($dispatcher->getActionName())->equals('hello');
+                    expect($dispatcher->getParams())->equals(array('World'));
+                    expect($dispatcher->getReturnedValue())->equals('Hello World!');
+                } catch (\Exception $e) {
+                    expect($e->getMessage())->equals('Dummy\MainTask handler class cannot be loaded');
```
Comment (sergeyklay) [inline tests/unit/Cli/ConsoleTest.php]: The same

File: tests/unit/Cli/ConsoleTest.php [hunk 8]
```diff
@@ -0,0 +1,490 @@
...
+                }
+            }
+        );
+    }
+
+    public function testArgumentOptions()
+    {
+        $this->specify(
+            "...",
```
Comment (sergeyklay) [inline tests/unit/Cli/ConsoleTest.php]: the same

File: tests/unit/Cli/DispatcherTest.php [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+/**
+ * \Phalcon\Test\Unit\Cli\DispatcherTest
+ * Tests the \Phalcon\Cli\Dispatcher component
+ *
+ * @copyright (c) 2011-2016 Phalcon Team
+ * @link      http://www.phalconphp.com
+ * @author    Andres Gutierrez <andres@phalconphp.com>
+ * @author    Nikolaos Dimopoulos <nikos@phalconphp.com>
+ * @package   Phalcon\Test\Unit
```
Comment (sergeyklay) [inline tests/unit/Cli/DispatcherTest.php]: `Phalcon\Test\Unit\Cli`

File: tests/unit/Cli/DispatcherTest.php [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+ * so that we can send you a copy immediately.
+ */
+class DispatcherTest extends UnitTest
+{
+    public function testDispatcher()
+    {
+        $this->specify(
+            "CLI Dispatcher doesn't work with typical parameters",
+            function () {
+                $di = new \Phalcon\DI\FactoryDefault\CLI();
```
Comment (sergeyklay) [inline tests/unit/Cli/DispatcherTest.php]: Use aliasing here. Di instead DI, Cli instead of CLI.

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
@SidRoberts Just ping me when it is ready for merging

File: tests/_data/micro/RestHandler.php [hunk 11]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+class RestHandler
```
Comment (sergeyklay) [inline tests/_data/micro/RestHandler.php]: We have `tests/_support/Helper` and `tests/_support/Module`. Why `tests/_data`?

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: tests/_data/micro/RestHandler.php (hunk 11 above)
Comment (sergeyklay) [inline tests/_data/micro/RestHandler.php]: ok

**Review by sergeyklay — APPROVED** [review]

**Comment by SidRoberts** [comment]
@sergeyklay That's all for now.

**Comment by sergeyklay** [comment]
Thanks!

**Pull request merged**